### PR TITLE
Add overrides for interface types and add CustomEvent types for testing.

### PR DESCRIPTION
**Pull request by jthoms1**
This PR will fix https://github.com/Microsoft/TypeScript/issues/14785 and replace https://github.com/Microsoft/TSJS-lib-generator/pull/304.

In order to override interfaces and add types this PR updates the TS.fsx and adds a new override type called `typedinterface`.  The naming convention was provided here https://github.com/Microsoft/TSJS-lib-generator/pull/304#issuecomment-341561311

This is my first time writing F# so please let me know if I need to make changes to the convention or if I missed an interfaces beyond Dictionaries or Interface Declarations.

### CODE REVIEW

File: inputfiles/overridingTypes.json [hunk 1]
```diff
@@ -1,5 +1,40 @@
 [
     {
+        "kind": "typedinterface",
```
Comment (mhegazy) [inline inputfiles/overridingTypes.json]: i do not think you need a new kind here.. just add a property `typeParameters`

Comment (jthoms1) [reply inputfiles/overridingTypes.json]: So would `kind` be `interface` and then `parameters` would become `typeParameters`?

**Comment by jthoms1** [comment]
Much better approach.  I have updated the code to reflect the requested changes.

File: inputfiles/overridingTypes.json [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+        "readonly": true,
+        "name": "detail",
+        "type": "T"
+    },
+    {
+        "kind": "method",
+        "interface": "CustomEvent",
+        "name": "initCustomEvent",
+        "signatures": [
+            "(typeArg: string, canBubbleArg: boolean, cancelableArg: boolean, detailArg: T): void"
```
Comment (mhegazy) [inline inputfiles/overridingTypes.json]: you need to include the name here. i.e. `initCustomEvent(typeArg: string, canBubbleArg: boolean, cancelableArg: boolean, detailArg: T)`

File: baselines/dom.generated.d.ts [hunk 3]
```diff
@@ -2374,9 +2374,9 @@ declare var CSSSupportsRule: {
     new(): CSSSupportsRule;
 };
 
-interface CustomEvent extends Event {
-    readonly detail: any;
-    initCustomEvent(typeArg: string, canBubbleArg: boolean, cancelableArg: boolean, detailArg: any): void;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: name of the method is missing.

File: baselines/dom.generated.d.ts [hunk 4]
```diff
@@ -2374,9 +2374,9 @@ declare var CSSSupportsRule: {
...
+interface CustomEvent<T = any> extends Event {
+    readonly detail: T;
+    (typeArg: string, canBubbleArg: boolean, cancelableArg: boolean, detailArg: T): void;
 }
 
 declare var CustomEvent: {
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: you need to change this one too, this is your entry point to an event.. it should be:

```ts
declare var CustomEvent: {
prototype: CustomEvent;
     new<T>(typeArg: string, eventInitDict?: CustomEventInit<T>): CustomEvent<T>;
 };
```

**Comment by jthoms1** [comment]
`initCustomEvent` is back in and I also added the type information to the constructor.

File: baselines/dom.generated.d.ts [hunk 5]
```diff
@@ -82,8 +82,8 @@ interface ConstrainVideoFacingModeParameters {
     ideal?: VideoFacingModeEnum | VideoFacingModeEnum[];
 }
 
-interface CustomEventInit extends EventInit {
-    detail?: any;
+interface CustomEventInit<T = any> extends EventInit {
+    detail: T;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: `detail` should still be optional? or is this intentional?

Comment (mhegazy) [reply baselines/dom.generated.d.ts]: looks good otherwise. thanks!

**Comment by jthoms1** [comment]
It is now optional. Thanks! Good learning experience I will likely have more PRs in the future.

**Comment by arjunyel** [comment]
@jthoms1 thank you for stepping up and seeing this through! I got stuck at the F# stuff and never made it through

**Comment by jthoms1** [comment]
@arjunyel No worries, I really wanted this. Your work started it! 😄

**Pull request merged**